inc/sitemaps/class-sitemaps.php

```diff
@@ -495,10 +495,16 @@ public static function get_last_modified_gmt( $post_types, $return_all = false )
 			$post_type_names = get_post_types( array( 'public' => true ) );
 
 			if ( ! empty( $post_type_names ) ) {
+				/**
+				 * Filter post status list for sitemap query for the post type.
+				 *
+				 * @param Array $post_status_names      Post status list, defaults to array( 'publish', 'inherit' ).
+				 */
+				$post_status_names = apply_filters( 'wpseo_sitemap_poststatus_lastmodified' , array( 'publish', 'inherit' ) );
 				$sql = "
 				SELECT post_type, MAX(post_modified_gmt) AS date
 				FROM $wpdb->posts
-				WHERE post_status IN ('publish','inherit')
+				WHERE post_status IN ('" . implode( "','", $post_status_names ) . "')
```

Comment:
The implode needs to be surrounded by a `esc_sql` to avoid sql injection.

Reply:
It has been added.